### PR TITLE
Expand also built-in variables in lsp_execute's command arguments

### PR DESCRIPTION
**Pull request by rchl**

### CODE REVIEW

**Review by predragnikolic — APPROVED** [review]

**Comment by predragnikolic** [comment]
Maybe also mention that addition in the docs? (it would be useful, to know that you can also use ST variables now)
https://sublimelsp.github.io/LSP/features/#execute-server-commands

File: docs/src/features.md [hunk 1]
```diff
@@ -40,14 +40,20 @@ Example:
...
 ```
-Note: `command_args` is optional depending on the `workspace/executeCommand` that are supported by the LSP server.
-The following variables will be expanded, but only if they are top-level array items and not within nested arrays or objects:
+
+Notes:
+
+ - the `session_name` is required and needs to match the server's key within the `clients` configuration object.
+ - the `command_args` is optional depending on the `workspace/executeCommand` that are supported by the LSP server.
+
+You can include special variables in the `command_args` array that will be automatically expanded. Supported variables include the Sublime's built-in ones (see the full list in [Build systems](http://www.sublimetext.com/docs/build_systems.html#variables) documentation) and also additional variables listed below. Note that the built-in variables will be expanded regardless of where they are in the array and also within nested arrays or objects while the variables listed below will only be expanded in the top-level array values and only if those values match exactly (will not match if they are sub-strings of values):
```
Comment (rwols) [inline docs/src/features.md:56]:
```suggestion
You can include special variables in the `command_args` array that will be automatically expanded. Supported variables include Sublime's built-in ones (see the full list in the [Build Systems](http://www.sublimetext.com/docs/build_systems.html#variables) documentation) as well as additional variables listed below. Note that the built-in variables will be expanded regardless of where they are in the array and also within nested arrays or objects while the variables listed below will only be expanded in the top-level array values and only if those values match exactly (will not match if they are sub-strings of values):
```

File: docs/src/features.md [hunk 2]
```diff
@@ -40,14 +40,20 @@ Example:
...
     }
   }
 ]
 ```
-Note: `command_args` is optional depending on the `workspace/executeCommand` that are supported by the LSP server.
-The following variables will be expanded, but only if they are top-level array items and not within nested arrays or objects:
+
+Notes:
+
+ - the `session_name` is required and needs to match the server's key within the `clients` configuration object.
```
[the remark this answers is not in the document]
Comment (predragnikolic) [reply docs/src/features.md:53]: "LSP-eslint" 🙂

[the remark this answers is not in the document]
Comment (rwols) [reply docs/src/features.md:53]: I know, but the average user doesn't know what a `clients` object is if he/she only downloaded helper packages.

[the remark this answers is not in the document]
Comment (predragnikolic) [reply docs/src/features.md:53]: But, that is not specified in the docs...

I believe LSP-metals has a different session name than I would expect. I will check it now

[the remark this answers is not in the document]
Comment (rchl) [reply docs/src/features.md:53]: I don't know if the average user will know how to set up custom execute commands anyway unless he/she gets the exact code to use.

I guess the options here are:
 - run `LSP: Troubleshoot server` to see the list of server
 - Enable debug messages and check the console when starting the server
 - Check the name in the lower-left corner of the status bar when server is running.

[the remark this answers is not in the document]
Comment (rwols) [reply docs/src/features.md:53]: Fair point.

**Review by rwols — APPROVED** [review]

File: docs/src/features.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (predragnikolic) [reply docs/src/features.md:53]: it is `metals`,

https://github.com/scalameta/metals-sublime/blob/c6a54d78dd0ffec49265dc914dd5c03556d05bb8/st4/__init__.py#L23

https://github.com/scalameta/metals-sublime/blob/c6a54d78dd0ffec49265dc914dd5c03556d05bb8/common/__init__.py#L57

```
class LspMetalsExecuteCommand(LspExecuteCommand):
    session_name = "metals"
```

----
So basically a `session_name` is something that is printed in the bottom left corner of ST status bar? Right?

[the remark this answers is not in the document]
Comment (rwols) [reply docs/src/features.md:53]: > So basically a session_name is something that is printed in the bottom left corner of ST status bar? Right?

Yes that's correct

**Pull request merged**